File: 05_Lists/ListViewStringArray/app/src/main/java/com/codingblocks/liststrarr/MainActivity.java
package com.codingblocks.liststrarr;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity {

    String[] names = new String[] {
            "Arnav",
            "Prateek",
            "Deepak",
            "Garima",
            "Rishab",
            "Harshit",
            "Aayush",
            "Rajesh",
            "Harpreet",
            "Nipun",
            "Anuj",
            "Rishabh",
            "Bhavya",
            "Apoorva"
    };
    ListView lvNames;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        lvNames = findViewById(R.id.lvNames);
        ArrayAdapter<String> namesAdapter = new ArrayAdapter<>(
                this,
                android.R.layout.simple_list_item_1,
                android.R.id.text1,
                names

        );
        lvNames.setAdapter(namesAdapter);
    }
}

File: W05_Database/app/src/main/java/com/codingblocks/todolist/MainActivity.kt
package com.codingblocks.todolist

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.ArrayAdapter
import com.codingblocks.todolist.db.MyDbHelper
import com.codingblocks.todolist.db.TodoTable
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    val todos = ArrayList<Todo>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val todoAdapter = ArrayAdapter<Todo>(
                this,
                android.R.layout.simple_list_item_1,
                android.R.id.text1,
                todos
        )
        val db = MyDbHelper(this).writableDatabase

        fun refreshTodoList () {
            val todoList = TodoTable.getAllTodos(db)
            todos.clear()
            todos.addAll(todoList)
            todoAdapter.notifyDataSetChanged()
        }
        lvTodos.adapter = todoAdapter
        refreshTodoList()

        btnAddTodo.setOnClickListener {
            val newTodo = Todo(
                    etNewTodo.text.toString(),
                    false
            )
            TodoTable.insertTodo(db, newTodo)
            refreshTodoList()
        }
    }
}
